docker/openemr/7.0.4/Dockerfile

```diff
@@ -1,26 +1,73 @@
-FROM alpine:3.21
 
-#Install dependencies and fix issue in apache
+# Below allows easy centralized setting of alpine version
+#  (note NOT meant to use this has an actually argument in building docker like the flex series does)
+ARG ALPINE_VERSION=3.22
+FROM alpine:${ALPINE_VERSION} AS base
+
+# Below allows easy centralized setting of PHP version
+#  (note NOT meant to use this has an actually argument in building docker like the flex series does)
+#  (note when update the PHP versiom, will also need build a updated php.ini file to match the new version)
+ARG PHP_VERSION=8.4
+ENV PHP_VERSION=${PHP_VERSION}
+ARG PHP_VERSION_ABBR=${PHP_VERSION//./}
+ENV PHP_VERSION_ABBR=${PHP_VERSION_ABBR}
+
+#Prepare to install dependencies
 RUN apk --no-cache upgrade
+
+# Install non-php packages
+RUN apk add --no-cache \
+    apache2 apache2-ssl apache2-utils apache2-proxy git perl mariadb-client mariadb-connector-c \
+    tar curl imagemagick nodejs npm certbot openssl openssl-dev dcron rsync shadow ncurses
+
+# Install PHP and its extensions
 RUN apk add --no-cache \
-    apache2 apache2-ssl apache2-utils apache2-proxy git php84 php84-tokenizer php84-ctype php84-session php84-apache2 \
-    php84-json php84-pdo php84-pdo_mysql php84-curl php84-ldap php84-openssl php84-iconv \
-    php84-xml php84-xsl php84-gd php84-zip php84-soap php84-mbstring php84-zlib \
-    php84-mysqli php84-sockets php84-xmlreader php84-redis php84-simplexml php84-xmlwriter php84-phar php84-fileinfo \
-    php84-sodium php84-calendar php84-intl php84-opcache php84-pecl-apcu php84-fpm \
-    perl mariadb-client mariadb-connector-c tar curl imagemagick nodejs npm \
-    certbot openssl openssl-dev dcron \
-    rsync shadow ncurses \
-    && sed -i 's/^Listen 80$/Listen 0.0.0.0:80/' /etc/apache2/httpd.conf
+    php${PHP_VERSION_ABBR} \
+    php${PHP_VERSION_ABBR}-tokenizer \
+    php${PHP_VERSION_ABBR}-ctype \
+    php${PHP_VERSION_ABBR}-session \
+    php${PHP_VERSION_ABBR}-apache2 \
+    php${PHP_VERSION_ABBR}-json \
+    php${PHP_VERSION_ABBR}-pdo \
+    php${PHP_VERSION_ABBR}-pdo_mysql \
+    php${PHP_VERSION_ABBR}-curl \
+    php${PHP_VERSION_ABBR}-ldap \
+    php${PHP_VERSION_ABBR}-openssl \
+    php${PHP_VERSION_ABBR}-iconv \
+    php${PHP_VERSION_ABBR}-xml \
+    php${PHP_VERSION_ABBR}-xsl \
+    php${PHP_VERSION_ABBR}-gd \
+    php${PHP_VERSION_ABBR}-zip \
+    php${PHP_VERSION_ABBR}-soap \
+    php${PHP_VERSION_ABBR}-mbstring \
+    php${PHP_VERSION_ABBR}-zlib \
+    php${PHP_VERSION_ABBR}-mysqli \
+    php${PHP_VERSION_ABBR}-sockets \
+    php${PHP_VERSION_ABBR}-xmlreader \
+    php${PHP_VERSION_ABBR}-redis \
+    php${PHP_VERSION_ABBR}-simplexml \
+    php${PHP_VERSION_ABBR}-xmlwriter \
+    php${PHP_VERSION_ABBR}-phar \
+    php${PHP_VERSION_ABBR}-fileinfo \
+    php${PHP_VERSION_ABBR}-sodium \
+    php${PHP_VERSION_ABBR}-calendar \
+    php${PHP_VERSION_ABBR}-intl \
+    php${PHP_VERSION_ABBR}-opcache \
+    php${PHP_VERSION_ABBR}-pecl-apcu \
+    php${PHP_VERSION_ABBR}-fpm
```

Comment:
I suggest alphabetizing these. It may seem trivial, but it's easier to track changes and eyeball if anything is missing if they're in a predictable order.